packages/ripple-binary-codec/src/shamap.ts

```diff
@@ -159,20 +159,22 @@ class ShaMapInner extends ShaMapNode {
    */
   addItem(index?: Hash256, item?: ShaMapNode, leaf?: ShaMapLeaf): void {
     assert.ok(index !== undefined)
-    const nibble = index.nibblet(this.depth)
-    const existing = this.branches[nibble]
-
-    if (existing === undefined) {
-      this.setBranch(nibble, leaf || new ShaMapLeaf(index, item))
-    } else if (existing instanceof ShaMapLeaf) {
-      const newInner = new ShaMapInner(this.depth + 1)
-      newInner.addItem(existing.index, undefined, existing)
-      newInner.addItem(index, item, leaf)
-      this.setBranch(nibble, newInner)
-    } else if (existing instanceof ShaMapInner) {
-      existing.addItem(index, item, leaf)
-    } else {
-      throw new Error('invalid ShaMap.addItem call')
+    if (index !== undefined) {
```

Comment:
was this change from dependabot?

Reply:
No, I made these changes - this was to deal with the new version of something not recognizing that `assert.ok(index !== undefined)` means that `index` is ok to use later on.

Reply:
(The Dependabot PR is separate & I'll close it once this gets merged :) )